tests/src/test/scala/cats/collections/HeapSpec.scala

```diff
@@ -19,4 +22,131 @@ class HeapSpec extends CatsSuite {
       heap.toList should be(exp.sorted)
 
     })
+
+  test("heapify is sorted") {
+    forAll { (set: Set[Int]) =>
+      val setList = set.toList
+      val heap = Heap.empty[Int].heapify(setList)
+
+      assert(heap.toList == setList.sorted)
+    }
+  }
+
+  def heapGen[A: Order](size: Int, agen: Gen[A]): Gen[Heap[A]] = {
+    val listA = Gen.listOfN(size, agen)
+    val startWith1 =
+      listA.map {
+        case Nil => Heap.empty[A]
+        case h :: tail => tail.foldLeft(Heap(h))(_.add(_))
+      }
+    val addOnly = listA.map(_.foldLeft(Heap.empty[A])(_.add(_)))
+    val heapify = listA.map(Heap.fromIterable(_))
+    // This one is recursive and with small probability can get quite deep
+    val addMoreAndRemove: Gen[Heap[A]] =
+      for {
+        extraSize <- Gen.choose(1, size + 1)
+        withExtra <- Gen.lzy(heapGen[A](size + extraSize, agen))
+      } yield (0 until extraSize).foldLeft(withExtra) { (h, _) => h.remove }
+    // we can also make smaller one and add to it:
+    val smallerAdd =
+      if (size > 0) {
+        for {
+          a <- agen
+          heap <- heapGen(size - 1, agen)
+        } yield heap + a
+      }
+      else Gen.const(Heap.empty[A])
+
+    Gen.frequency((2, addOnly), (3, startWith1), (5, heapify), (1, addMoreAndRemove), (1, smallerAdd))
+  }
+
+  implicit def arbHeap[A: Arbitrary: Order]: Arbitrary[Heap[A]] =
+    Arbitrary {
+      Gen.choose(0, 10000).flatMap(heapGen[A](_, Arbitrary.arbitrary[A]))
```

Comment:
This should probably use the `sized` combinator. If we want large test sizes, then we can also increase that in the config.